File: ios/RNBilAdBanner.m
//
//  RNBilAdBanner.m
//  react-native-bilmobileads
//
//  Created by HNL_MAC on 27/01/2021.
//

#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNBilAdBanner, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(adUnitId, NSString)
RCT_EXPORT_VIEW_PROPERTY(onNativeEvent, RCTBubblingEventBlock);

RCT_EXTERN_METHOD(load:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(show:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(hide:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(destroy:(nonnull NSNumber *)node)

@end

File: ios/RNBilAdBanner.swift
//
//  RNBilAdBanner.swift
//  react-native-bilmobileads
//
//  Created by HNL_MAC on 27/01/2021.
//

@objc(RNBilAdBanner)
class RNBilAdBanner: RCTViewManager {
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func view() -> UIView! {
        return RNAdBannerView()
    }
    
    @objc func load(_ node: NSNumber){
        DispatchQueue.main.async {
            let banner = self.bridge.uiManager.view(forReactTag: node) as! RNAdBannerView
            banner.load()
        }
    }
    
    @objc func show(_ node: NSNumber){
        DispatchQueue.main.async {
            let banner = self.bridge.uiManager.view(forReactTag: node) as! RNAdBannerView
            banner.show()
        }
    }
    
    @objc func hide(_ node: NSNumber){
        DispatchQueue.main.async {
            let banner = self.bridge.uiManager.view(forReactTag: node) as! RNAdBannerView
            banner.hide()
        }
    }
    
    @objc func destroy(_ node: NSNumber){
        DispatchQueue.main.async {
            let banner = self.bridge.uiManager.view(forReactTag: node) as! RNAdBannerView
            banner.destroy()
        }
    }
}
